### ISSUE TITLE
The Tax calculator

### ISSUE THREAD

**Issue by PaulineTK** [stevermeister/dutch-tax-income-calculator-old#73]
Hi 
From forein workers i have recieved question about the tax calculated here and the tax on their payslip since there is a difference between them. 
How do you calculate the tax payable to the tax office? 

For the payslips we have to use the montly tabel provides bij the tax office, the match with the payslips.

Greetings

Pauline

**Comment by stevermeister** [comment]
it's open source, so you can check the logic - 
https://github.com/stevermeister/dutch-tax-income-calculator/blob/master/src/calc/calc.component.js

it goes box-by-box, base on parameters from here: 
https://github.com/stevermeister/dutch-tax-income-calculator/blob/master/src/data.json

> the montly tabel provides bij the tax office

not sure that I understand you. 
All criteria are set from the beginning of the year and final result could be calculated by exact formula. What kind of table your are talking?

**Comment by PaulineTK** [comment]
Thank you for the links, I do not understand the first for I do not know
programming language, the second makes more sense.
I have to make some more calculations to see how this all works. You work
from full year as far as I understand this.

The tax table I am talkong about is found here:
https://download.belastingdienst.nl/belastingdienst/dl/rekenhulpen/loonheffing/2018/v01/pdf/wit_mnd_std_01012018.pdf

This gives the exact amount of tax to be paid per month due to you income
per month. This we have to use in the payroll administration.



Op do 27 sep. 2018 om 14:52 schreef Stepan Suvorov <notifications@github.com

**Comment by stevermeister** [comment]
> You work from full year as far as I understand this.

yes, correct.


> The tax table I am talkong about is found here:

But this tax table is not just set of magic numbers they are calculated by specific formulas that I've implemented in my application.

> This we have to use in the payroll administration.

But it's not all. How would now the result from the table if person has tax deduction benefit (for example 30% ruling)?

I think it's better to have good formula with up-to-date coefficients to be more flexible with it.

**Comment by PaulineTK** [comment]
I do understand that you need to be flexible, however I do not want to
explain that what is calculated on your site is different from what is
actual on the payslips.
For normal wages the first two brackets are too low for wages that do not
have the 30% ruling.
When I tried it in my own wage it also was far to low on the tax to be
paid.

First bracket (whole year) = t/m € 20142 - 36.55% tax
second bracket  = van € 20143 t/m € 33994 - 40.85% tax
third bracket = van € 33995 t/m € 68507  = 40.85% tax
forth bracket = over € 68507 - 51.95% tax



Op zo 30 sep. 2018 om 10:18 schreef Stepan Suvorov <notifications@github.com

**Comment by stevermeister** [comment]
as you see from here:
https://github.com/stevermeister/dutch-tax-income-calculator/blob/master/src/data.json#L110

I have different percentage for this 2 positions: 

>First bracket (whole year) = t/m € 20142 - 36.55% tax
>second bracket  = van € 20143 t/m € 33994 - 40.85% tax

could I ask you to send me the link please with description?

**Comment by PaulineTK** [comment]
Yes offcourse.
Link to the brackets for 2018 normal wages are here
https://www.belastingdienst.nl/wps/wcm/connect/bldcontentnl/belastingdienst/prive/inkomstenbelasting/heffingskortingen_boxen_tarieven/boxen_en_tarieven/overzicht_tarieven_en_schijven/u-hebt-in-2018-de-aow-leeftijd-nog-niet-bereikt


Op ma 1 okt. 2018 om 11:17 schreef Stepan Suvorov <notifications@github.com

**Comment by stevermeister** [comment]
thank you very much for the information!

I'm using numbers from here
https://www.expatax.nl/tax-rates-2018

so total rate should work the same.

Do you know how to do the calculations?
It would be great if you could give me calculations for your case that did not match mine that I can check it step by step.

**Comment by PaulineTK** [comment]
Thank you, now it makes more sence for the insurance premium is for all
employees, you cannot not pay it. But then employee should add it to the
amount toe be paid and that is where I also went wrong for I did not
recognise it for what it is.
And yes I do know how to make the calculation for I am a payroll
administrator.
But now if I look to all the tax instead of the payroll tax only, it is
only a difference of 58 cent so that would be close enough for forcasting.

The social premiums were confusing for they are not visible on the dutch
payslips anymore, only for the employer they are made specific and thus
visible




Op ma 1 okt. 2018 om 11:37 schreef Stepan Suvorov <notifications@github.com
